snuba/datasets/discover.py

```diff
@@ -234,6 +240,7 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
 
         return [
             DatasetSelector(discover_source, self.__transactions_columns),
+            PreWhereProcessor(),
```

Comment:
Since the Prewhere candidates are part of the tableSource, I suspect we could have a list of default query processors that are always applied (like this one), no matter on the dataset. For now it is still per dataset.

Reply:
I suspect this should eventually be part of a different abstraction than this level of query processing (elaborated in a different comment below) but in general I agree that it should be something that comes along "for free" when using ClickHouse MergeTree-backed storage engines.